src/ngraph/runtime/backend_manager.cpp

```diff
@@ -107,13 +113,13 @@ shared_ptr<runtime::Backend> runtime::BackendManager::create_backend(const std::
             stringstream ss;
             ss << "Backend '" << type << "' not registered. Error:";
 #ifndef _WIN32
-            ss << dlerror();
+            ss << DLERROR();
 #endif
             throw runtime_error(ss.str());
         }
-
+#ifndef NGRAPH_STATIC_LIB_ENABLE
```

Comment:
I think you need to move this up a few lines. You are opening the shared library but then never using it. I think your intent is to not open the shared library, right?